de1plus/profile.tcl

```diff
@@ -520,41 +520,110 @@ namespace eval ::profile {
         return $profile_type
     }
         
-    # Based on proc load_settings_vars, but only loads profile variables, and returns a list with the profile (which can be
-    # coerced to an array) instead of loading the profile into the global settings. The source file can also be a shot 
-    # file, and only the profile-related vars will be loaded from it, plus any variables given in list $extra_vars.
-    proc read_legacy { profile_path {extra_vars {}} } {
-        if { [file pathtype $profile_path] eq "relative" && [file dirname $profile_path] eq "." } {
-            set profile_path "[homedir]/profiles/[file tail $profile_path]"
+    proc profile_type_text { profile_type {beverage_type {}} } {
+        switch [fix_profile_type $profile_type] \
+        settings_2a {
+            set type "Pressure"
+        } settings_2b {
+            set type "Flow"
+        } settings_2c {
+            set type "Advanced"
+        } default {
+            set type "Unknown"
         }
-        if { [file extension $profile_path] eq "" } {
-            append profile_path ".tcl"
+        
+        switch [lindex $beverage_type 0] \
+            pourover {
+                set beverage_type "pour-over "
+            } tea_portafilter {
+                set beverage_type "tea "
+            } {} {
+                set beverage_type ""
+            } default {
+                set beverage_type "$beverage_type "
+            }
+        
+        return "${type} ${beverage_type}profile"
+    }
+    
+    proc find_file { filename {legacy 0} } {
+        if { [string is true $legacy] } {
+            set folder "profiles"
+            set ext ".tcl"
+        } else {
+            set folder "profiles_v2"
+            set ext ".json"
         }
-        if { ![file exists $profile_path] } {
-            msg -WARNING [namespace current] load_profile: "profile file '$profile_path' not found"
-            return
+
+        if { [file extension $filename] eq "" || [file extension $filename] ne $ext } {
+            append filename $ext
         }
         
-        msg -INFO [namespace current] read_profile: "reading file '$profile_path'"
+        if { [file pathtype $filename] eq "relative" && [file dirname $filename] eq "." } {
```

Comment:
shouldn't we handle ./ and ./profiles as well (at least in theory)?

Reply:
"./" is handled, because both `[file dirname "./abc.tcl"]` and `[file dirname "abc.tcl"]` return ".".

"./profiles" is not handled, true. I didn't expect that type of input normally, only either filenames or full paths.

Reply:
okay, can we add a warning then? if is relative && ![file dirname $filename] eq "."

Reply:
But you will already get the warning that the file does not exists: "de1plus/profiles/profiles/default.tcl"

Reply:
Added case to handle "./profiles" in a new commit. Plus, spotted a bug with a hardcoded `profiles` instead of `$folder`.